src/machines/java/com/enderio/machines/common/blockentity/PowerBufferBlockEntity.java

```diff
@@ -0,0 +1,240 @@
+    int maxInput = 0;
+    protected IntegerNetworkDataSlot inputDataSlot;
+
+    int maxOutput = 0;
+    protected IntegerNetworkDataSlot outputDataSlot;
+
+    // Stores textbox value before being converted to energy value
+    protected StringNetworkDataSlot inputTextDataSlot;
+    String inputTextValue = "0";
+    protected StringNetworkDataSlot outputTextDataSlot;
+    String outputTextValue = "0";
+
+    //todo: energy balancing
+    public static final QuadraticScalable CAPACITY = new QuadraticScalable(CapacitorModifier.ENERGY_CAPACITY, MachinesConfig.COMMON.ENERGY.POWER_BUFFER_CAPACITY);
+    public static final QuadraticScalable USAGE = new QuadraticScalable(CapacitorModifier.ENERGY_USE, MachinesConfig.COMMON.ENERGY.POWER_BUFFER_USAGE);
+
+    public PowerBufferBlockEntity(BlockEntityType<?> type, BlockPos worldPosition, BlockState blockState) {
+        super(EnergyIOMode.Both, CAPACITY, USAGE, type, worldPosition, blockState);
+
+        inputDataSlot = new IntegerNetworkDataSlot(this::getMaxInput, input -> maxInput = input);
+        outputDataSlot = new IntegerNetworkDataSlot(this::getMaxOutput, output -> maxOutput = output);
+        addDataSlot(inputDataSlot);
+        addDataSlot(outputDataSlot);
+
+        inputTextDataSlot = new StringNetworkDataSlot(this::getMaxInputText, input -> inputTextValue = input);
+        outputTextDataSlot = new StringNetworkDataSlot(this::getMaxOutputText, output -> outputTextValue = output);
+        addDataSlot(inputTextDataSlot);
+        addDataSlot(outputTextDataSlot);
+    }
+
+    @Override
+    public @Nullable MachineInventoryLayout getInventoryLayout() {
+        return MachineInventoryLayout
+            .builder()
+            .capacitor()
+            .build();
+    }
+
+    @Override
+    protected boolean isActive() {
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public AbstractContainerMenu createMenu(int containerId, Inventory playerInventory, Player player) {
+        return new PowerBufferMenu(this, playerInventory, containerId);
+    }
+
+    public void setMaxOutput(int pMaxOutput) {
+        if (level != null && level.isClientSide) {
+            clientUpdateSlot(outputDataSlot, pMaxOutput);
+        } else this.maxOutput = pMaxOutput;
+    }
+
+    public void setMaxInput(int pMaxInput) {
+        if (level != null && level.isClientSide) {
+            clientUpdateSlot(inputDataSlot, pMaxInput);
+        } else this.maxInput = pMaxInput;
+    }
+
+    public void setMaxInputText(String maxInputText) {
+        if (level != null && level.isClientSide) {
+            clientUpdateSlot(inputTextDataSlot, maxInputText);
+        } this.inputTextValue = maxInputText;
+    }
+
+    public void setMaxOutputText(String maxOutputText) {
+        if (level != null && level.isClientSide) {
+            clientUpdateSlot(outputTextDataSlot, maxOutputText);
+        } this.outputTextValue = maxOutputText;
+    }
```

Comment:
There is no point in syncing the text. It should be completely client sided. Only the server needs to know the final Integer.